src/content-helper/editor-sidebar/related-posts/component.tsx

```diff
@@ -189,43 +208,55 @@ export const RelatedPostsPanel = (): React.JSX.Element => {
 	 */
 	const onPeriodChange = ( selection: string ) => {
 		if ( isInEnum( selection, Period ) ) {
+			const updatedPeriod = selection as Period;
+
 			setSettings( {
 				RelatedPosts: {
 					...settings.RelatedPosts,
-					Period: selection as Period,
+					Period: updatedPeriod,
 				},
 			} );
-			Telemetry.trackEvent( 'related_posts_period_changed', { period: selection } );
+			Telemetry.trackEvent( 'related_posts_period_changed', { period: updatedPeriod } );
+
+			fetchPosts( updatedPeriod, metric, filters, FETCH_RETRIES );
 		}
 	};
 
-	useEffect( () => {
```

Comment:
`fetchPosts()` was previously contained in a `useEffect()`. The problem with this setup is that each time the component was unmounted and remounted (by closing the wp-parsely pane or opening a different pane), this logic would retrigger.

`fetchPosts()` is now called in event handlers rather than relying on a stable mount.